### DISCUSSION TITLE
@vue/compiler-sfc@2.7 should support esm-browser

### DISCUSSION THREAD

**Discussion by 2392645016** [2392645016/vue#8] [Q&A] [answered]
Version
2.7.7

Steps to reproduce
打包 @vue/compiler-sfc@2.7 时报node模块不存在

What is expected?
打包 @vue/compiler-sfc@2.7 时不报错

What is actually happening?
打包 @vue/compiler-sfc@2.7 时报node模块不存在

**Comment by 202780181** [comment, marked as answer]
<template>
  <div>
    <p>I'm parent.</p>

    <p>This can work: {{right}} -> (<ChildPage/>)</p>
    <p>This can't work: {{wrong}} -> (<child-page/>)</p>

    <p>Because I coded 'const childPage = ref(null)'. If I delete it, they both can show.</p>
  </div>
</template>
<script setup>
import ChildPage from './ChildPage.vue'
import { ref } from 'vue'

const right = ref(`<ChildPage/>`)
const wrong = ref(`<child-page/>`)

const childPage = ref(null) // If I delete it, they both can show.
</script>
<style></style>

**Reply by 202780181** [reply]
Kebab-case component can be rendered.
